src/Microsoft.SqlTools.ServiceLayer/LanguageExtensibility/ExternalLanguageOperations.cs

```diff
@@ -0,0 +1,343 @@
+        public virtual List<ExternalLanguage> GetLanguages(IDbConnection connection)
+        {
+            return GetLanguages(connection, null);
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="connection"></param>
+        /// <param name="languageName"></param>
+        /// <returns></returns>
+        public virtual ExternalLanguage GetLanguage(IDbConnection connection, string languageName)
+        {
+            List<ExternalLanguage> result = GetLanguages(connection, languageName);
+            if (result != null && result.Any())
+            {
+                return result.First();
+            }
+            return null;
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="connection"></param>
+        /// <param name="language"></param>
+        public virtual void UpdateLanguage(IDbConnection connection, ExternalLanguage language)
+        {
+            if (string.IsNullOrWhiteSpace(language?.Name))
+            {
+                throw new LanguageExtensibilityException($"Invalid language. name: {language?.Name}");
+            }
+
+            Dictionary<string, object> parameters = new Dictionary<string, object>();
+            ExternalLanguage currentLanguage = GetLanguage(connection, language.Name);
+            if (currentLanguage == null)
+            {
+                ExecuteNonQuery(connection, GetCreateScript(language, parameters), parameters);
+            }
+            else
+            {
+                foreach (var content in language.Contents)
+                {
+                    var currentContent = currentLanguage.Contents.FirstOrDefault(x => x.PlatformId == content.PlatformId);
+                    if (currentContent != null)
+                    {
+                        ExecuteNonQuery(connection, GetUpdateScript(language, content, parameters, ContentModifyType.Modify), parameters);
+                    }
+                    else
+                    {
+                        ExecuteNonQuery(connection, GetUpdateScript(language, content, parameters, ContentModifyType.Add), parameters);
+
+                    }
+                }
+                foreach (var currentContent in currentLanguage.Contents)
+                {
+                    var content = language.Contents.FirstOrDefault(x => x.PlatformId == currentContent.PlatformId);
+                    if (content == null)
+                    {
+                        ExecuteNonQuery(connection, GetUpdateScript(language, currentContent, parameters, ContentModifyType.Remove), parameters);
+
+                    }
+                }
+            }
+        }
+
+        public virtual void DeleteLanguage(IDbConnection connection, string languageName)
+        {
+            if (string.IsNullOrWhiteSpace(languageName))
+            {
+                throw new LanguageExtensibilityException($"Invalid language name. name: {languageName}");
```

Comment:
>languageName [](start = 89, length = 12)

nit: This is never going to print anything useful for language name since it's always empty or null